relayer-cli/src/commands/config/auto.rs

```diff
@@ -0,0 +1,93 @@
+use std::path::PathBuf;
+
+use abscissa_core::clap::Parser;
+use abscissa_core::{Command, Runnable};
+use chain_registry::{error::RegistryError, relayer_config::hermes_config};
+
+use crate::conclude::Output;
+
+use futures::{stream::FuturesUnordered, StreamExt};
+
+use ibc_relayer::config::{store, ChainConfig, Config};
+
+use tokio::runtime::Builder;
+
+#[derive(Clone, Command, Debug, Parser, PartialEq)]
+pub struct AutoCmd {
+    #[clap(
+        long = "path",
+        short = 'p',
+        required = true,
+        value_name = "PATH",
+        help_heading = "REQUIRED",
+        help = "Path to the configuration file"
+    )]
+    path: PathBuf,
+
+    #[clap(
+        long = "chains",
+        required = true,
+        multiple = true,
+        value_name = "CHAIN_1 CHAIN_2...",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chains to include in the config"
+    )]
+    chain_ids: Vec<String>,
+
+    #[clap(
+        long = "keys",
+        required = true,
+        multiple = true,
+        value_name = "KEY_CHAIN_1 KEY_CHAIN_2...",
+        help_heading = "REQUIRED",
+        help = "Key names to include in the config"
+    )]
+    keys: Vec<String>,
+}
+
+async fn get_chain_configs(
+    chain_names: &[String],
+    keys: &[String],
+) -> Vec<Result<ChainConfig, RegistryError>> {
+    let futures: FuturesUnordered<_> = chain_names
```

Comment:
Let's add an `assert!` to enforce the invariant used in `zip` below:

```suggestion
) -> Vec<Result<ChainConfig, RegistryError>> {
    assert_eq!(chain_names.len(), keys.len());

    let futures: FuturesUnordered<_> = chain_names
```